### PR TITLE
Волков Павел - Лабораторная работа №2

### PR DESCRIPTION
**Pull request by Snewar79**

### CODE REVIEW

**Comment by Snewar79** [comment]
@owerbat
@EOKruglov

**Comment by EOKruglov** [comment]
👍

File: modules/bit_array/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Declare variables for binaries' names
+get_filename_component(DIR_NAME ${CMAKE_CURRENT_LIST_DIR} NAME)
+set(MODULE      "${DIR_NAME}")
+set(LIBRARY     "lib_${MODULE}")
+set(TESTS       "test_${MODULE}")
+
+# Include directory with public headers
+include_directories("./")
```
Comment (alexbatashev) [inline modules/bit_array/CMakeLists.txt]: CMAKE_CURRENT_SOURCE_DIR

File: modules/bit_array/test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+set(target ${TESTS})
+
+file(GLOB srcs "*.cpp")
+set_source_files_properties(${srcs} PROPERTIES
+    LABELS "${MODULE};Test")
+
+# VS2012 doesn't support correctly the tuples yet,
```
Comment (alexbatashev) [inline modules/bit_array/test/CMakeLists.txt]: You don’t need this

File: modules/bit_array/include/bit_array.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+    void SetBit(int bit_number, bool value);
+
+    bool GetBit(int bit_number);
+
+    std::string print_string();
+
+ private:
+    int *memory;
```
Comment (alexbatashev) [inline modules/bit_array/include/bit_array.h]: Use vector

File: modules/bit_array/src/bit_array.cpp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    if (bit_count <= 0)
+        throw std::string("Can't create negative and zero array");
+
+    memory_size = bit_count / (8 * sizeof(int));
+
+    if (bit_count % (8 * sizeof(int)) != 0)
+        memory_size += 1;
+
+    memory = static_cast<int*>(malloc(memory_size * sizeof(int)));
```
Comment (alexbatashev) [inline modules/bit_array/src/bit_array.cpp]: Or at least new

Comment (Snewar79) [reply modules/bit_array/src/bit_array.cpp]: Ok, I will use new

File: modules/bit_array/test/CMakeLists.txt (hunk 2 above)
Comment (Snewar79) [reply modules/bit_array/test/CMakeLists.txt]: Fixed

File: modules/bit_array/CMakeLists.txt (hunk 1 above)
Comment (Snewar79) [reply modules/bit_array/CMakeLists.txt]: Fixed

File: modules/bit_array/src/bit_array.cpp (hunk 4 above)
Comment (Snewar79) [reply modules/bit_array/src/bit_array.cpp]: Fixed
Now using c++ style with new

**Pull request merged**